bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php

```diff
@@ -2500,4 +2500,121 @@ protected function processAvailableFieldDefinitions($fds, &$firstOne, &$commonFi
+        
+        try {
+            
+            $configName = $configData['shareSettings']['configName'];
+            
+            if(empty($configName)){
+                $configName = date("YmdHis")."_".$classId."_configuration";
+            }
+            
+            $jsonResponse = new JsonResponse(json_encode($configData), 200, [
+                'Content-Disposition' => 'attachment; filename="'.$configName.'.json"'
+            ], true);
+            
+            return $jsonResponse;
+            
+        } catch (\Exception $e) {
+            throw new \Exception("Error retrieving import configuration - ".$e->getMessage());
+        }
+    }
+    
+    /**
+     * @Route("/import-json-config", methods={"POST"})
+     *
+     * @param Request $request
+     * @param ImportService $importService
+     *
+     * @return JsonResponse
+     */
+    public function importJsonConfigAction(Request $request, ImportService $importService)
+    {
+        $importConfigId = $request->get('importConfigId');
+        
+        $tmpName = $_FILES['Filedata']['tmp_name'];
+        $json = file_get_contents($tmpName);
+
+        $configData = json_decode($json, true);
+
+        $dataFields = $configData['dataFields'];
+        $targetFields = $configData['targetFields'];
+        $selectedGridColumns = $configData['selectedGridColumns'];
+        $resolverSettings = $configData['resolverSettings'];
+        $shareSettings = $configData['shareSettings'];
+        $dialect = json_decode(json_encode($configData['csvSettings']), false);
+        
+        $success = true;
+        $classId = $request->get('classId');
+        
+        $file = PIMCORE_SYSTEM_TEMP_DIRECTORY . '/import_' . $request->get('importId');
+
+        $originalFile = $file . '_original';
+        // determine type
+        if (empty($dialect)) {
+            $dialect = Tool\Admin::determineCsvDialect($originalFile);
+        }
+
+        /**
+         * Reload data form original CSV to properly refresh 
+         * the data preview on the import interface
+         */
+        $count = 0;
+        $data = [];
+        if (($handle = fopen($originalFile, 'r')) !== false) {
+            while (($rowData = fgetcsv($handle, 0, $dialect->delimiter, $dialect->quotechar, $dialect->escapechar)) !== false) {
+                $tmpData = [];
+
+                foreach ($rowData as $key => $value) {
+                    $tmpData['field_' . $key] = $value;
+                }
+
+                $tmpData['rowId'] = $count + 1;
+                $data[] = $tmpData;
+
+                $count++;
+
+                /**
+                 * Reached the number or rows for the preview
+                 */
+                if ($count > 18) {
+                    break;
+                }
+            }
+            fclose($handle);
+        }
+        
+        $availableConfigs = $this->getImportConfigs($importService, $this->getAdminUser(), $classId);
```

Comment:
why do we use the admin user here ? shouldn't it be the user that triggers the import action ?

Reply:
I'm trying to use the $this->getUser() function, but i got this exception.

![Exception](https://user-images.githubusercontent.com/36894331/78028760-c6a4d500-735f-11ea-92aa-d9335acf9b0f.png)

The exception is thrown on the $importService->getSharedImportConfigs($user, $classId) function.

Which is the correct function to call to get the current user?

Reply:
\Pimcore\Tool\Admin::getCurrentUser() I believe

Reply:
Thank you, fixed